upload/src/addons/SV/CustomFieldPerms/XF/Entity/UserField.php

```diff
@@ -1,10 +1,30 @@
 <?php
 
-namespace SV\CustomFieldPerms\XenForo\Entity;
+namespace SV\CustomFieldPerms\XF\Entity;
+
+use SV\CustomFieldPerms\Setup;
 
 /*
- * Extends \XenForo\Entity\UserField
+ * Extends \XF\Entity\UserField
  */
 class UserField extends XFCP_UserField
 {
+
+	/**
+	 * Insert additional columns into UserField
+	 *
+	 * @param \XF\Mvc\Entity\Structure $structure
+	 *
+	 * @return \XF\Mvc\Entity\Structure
+	 */
+	public static function getStructure(\XF\Mvc\Entity\Structure $structure) {
+		$structure = parent::getStructure($structure);
+
+		foreach (Setup::$tables1['xf_user_field'] as $column => $details) {
+			$structure->columns += [$column => ['type'=>$details['entity_type'], 'default' => $details['entity_default']]];
```

Comment:
Style wise would be a better solution to just directly set the column `$structure->columns[$column] = ...` compared to appending another array

Reply:
OK